src/com/hms/flexyosisoftconnector/PayloadBuilder.java

```diff
@@ -0,0 +1,337 @@
+   *
+   * @param timestamp The time should be in format yyyy-mm-ddThh:mm:ssZ.
+   * @return the payload segment as a string
+   */
+  public static String endPayload(String timestamp) {
+    String payload = "],\"lastSynchroDate\": \"" + timestamp + "\"}";
+    return payload;
+  }
+
+  public static String startOMFDataMessage() {
+    String payload = "[";
+    return payload;
+  }
+
+  /**
+   * * This constructs the start of a new container
+   *
+   * @param the payload segment as a string
+   */
+  public static String addContainerStartToOMFDataMessage(String tagName) {
+    String payload = "";
+    payload += "{";
+
+    // each tag's container id is set to the tag's name
+    payload += "\"containerid\": \"" + tagName + "\"";
+    payload += ",";
+    payload += "\"values\": [";
+
+    return payload;
+  }
+
+  /**
+   * This constructs a new data point omf message
+   *
+   * @return return the JSON payload for a new OMF data point
+   */
+  public static String addPointToOMFDataMessage(
+      String tagValue, String timestamp, boolean isFirstPoint) {
+    String payload = "";
+
+    if (!isFirstPoint) {
+      payload += ",";
+    }
+
+    payload += "{";
+    payload +=
+        " \"timestamp\": \""
+            + timestamp
+            + ".000Z\","; // should move the timezone part into the timestamp string
+    payload += " \"tagValue\": \"" + tagValue + "\"";
+    payload += "}";
+
+    return payload;
+  }
+
+  /**
+   * call after finished adding all data points to close values array in json
+   *
+   * @return return the JSON payload for the end of an OMF container message.
+   */
+  public static String addContainerEndToOMFDataMessage() {
+    String payload =
+        "]}"; // closes off the end of the container specific portion of the data message
+    return payload;
+  }
+
+  /**
+   * call this when finished building up the OMF message
+   *
+   * @return return the JSON payload for the end of an OMF data message.
+   */
+  public static String endOMFDataMessage() {
+
+    String payload = "]";
+    return payload;
+  }
+
+  /**
+   * Setup the format for how data from this device will be stored with omf
+   *
+   * @return returns the JSON payload to construct a new Type message.
+   */
+  public static String getTypeBody() {
+
+    String flexyName = getFlexyName();
+    String payload = "[{";
+    payload += "\"id\": \"HMS-type-" + flexyName + "\",";
+    payload += "\"classification\": \"dynamic\",";
+    payload += "\"type\": \"object\",";
+    payload += "\"properties\": {";
+    payload += "\"timestamp\": {";
+    payload += "\"type\": \"string\",";
+    payload += "\"format\": \"date-time\",";
+    payload += "\"isindex\": true";
+    payload += "},";
+    payload += "\"tagValue\": {";
+    payload += "\"type\": \"string\",";
+    payload += "\"description\": \"Ewon Flexy's tag value stored as a string\"";
+    payload += "}";
+    payload += "}";
+    payload += "}]";
```

Comment:
may be neater with a [StringBuffer](https://www2.cs.duke.edu/csed/java/jdk1.4.2/docs/api/java/lang/StringBuffer.html)

Reply:
I think it would be the same either way, but I could combine lines if that would add to the clarity at all. just appending instead of +=

Reply:
It's fine either way, although in loops and large string operations, using a StringBuffer can produce better performance than direct string concatenation.

Reply:
Its best to avoid unnecessary reallocation